src/vs/workbench/browser/parts/editor/editorDropTarget.ts

```diff
@@ -310,25 +310,37 @@ class DropOverlay extends Themable {
 			const data = this.editorTransfer.getData(DraggedEditorIdentifier.prototype);
 			if (Array.isArray(data)) {
 				const draggedEditor = data[0].identifier;
-				const targetGroup = ensureTargetGroup();
 
 				const sourceGroup = this.accessor.getGroup(draggedEditor.groupId);
 				if (sourceGroup) {
-					if (sourceGroup === targetGroup) {
-						return;
+					const copyEditor = this.isCopyOperation(event, draggedEditor);
+					let targetGroup: IEditorGroupView | undefined = undefined;
+
+					// Optimization: if we move the last editor of an editor group
+					// and we are configured to close empty editor groups, we can
+					// rather move the entire editor group according to the direction
+					if (this.editorGroupService.partOptions.closeEmptyGroups && sourceGroup.count === 1 && typeof splitDirection === 'number' && !copyEditor) {
+						targetGroup = this.accessor.moveGroup(sourceGroup, this.groupView, splitDirection);
```

Comment:
@joaomoreno fyi this is the optimization to leverage grid moving here in the case of:
* 1 editor being moved in a group of 1 editor
* configuration to not keep empty groups opened
* this is not a copy operation